src/region/regionUtil.ts

```diff
@@ -29,6 +44,98 @@ export const centerRegion = (shape: Shape): Position => {
     };
 };
 
+export const getPoints = (shape: Shape): [Point, Point, Point, Point] => {
+    const { height, width, x, y } = shape;
+
+    // Define the points from x,y and then in a clockwise fashion
+    // p1      p2
+    // +-------+
+    // |       |
+    // +-------+
+    // p4      p3
+    const p1 = { x, y };
+    const p2 = { x: x + width, y };
+    const p3 = { x: x + width, y: y + height };
+    const p4 = { x, y: y + height };
+
+    return [p1, p2, p3, p4];
+};
+
+export function selectTransformationPoint(shape: Shape, rotation: number): Point | null {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [p1, p2, p3, p4] = getPoints(shape);
+
+    // Determine which point will be the new x,y (as defined as the top left point) after rotation
+    // If -90deg: use p2
+    // If -180deg: use p3
+    // If -270deg: use p4
+    switch (rotation) {
+        case ROTATION_ONCE_DEG:
+            return p2;
+        case ROTATION_TWICE_DEG:
+            return p3;
+        case ROTATION_THRICE_DEG:
+            return p4;
+        default:
+            return null;
+    }
+}
+
+// Determines the translation needed to anchor the bottom left corner of the
+// coordinate space at the (0, 0) origin after rotation.
+export function selectTranslation(coordinateSpace: CoordinateSpace, rotation: number): Translation {
+    const { height, width } = coordinateSpace;
+
+    switch (rotation) {
+        case ROTATION_ONCE_DEG:
+            return { dx: height };
+        case ROTATION_TWICE_DEG:
+            return { dx: width, dy: height };
+        case ROTATION_THRICE_DEG:
+            return { dy: width };
+        default:
+    }
+
+    return { dx: 0, dy: 0 };
+}
+
+export function getTransformedShape(
+    shape: Shape,
+    rotation: number,
+    coordinateSpace: CoordinateSpace = { height: 100, width: 100 },
+): Shape {
+    const { height: shapeHeight, width: shapeWidth } = shape;
+    const { height: spaceHeight, width: spaceWidth } = coordinateSpace;
+    const isInverted = rotation % 180 === 0;
+    const isNoRotation = rotation % 360 === 0;
+    const translation = selectTranslation(coordinateSpace, rotation);
+    const point = selectTransformationPoint(shape, rotation);
+
+    if (isNoRotation || !point) {
+        return shape;
+    }
+
+    // To transform from shape with 0 rotation to provided rotation:
+    // 1. Invert y-axis to convert from web to mathematical coordinate system
```

Comment:
Is there a way to modify the formula we use to calculate rotation to work with the DOM coordinate system, instead? The conversion to/from seems to add a lot of complexity to the situation.

Reply:
I'm guessing there might be a one shot matrix transformation that could take care of this, which I can look into for a follow up, but for now, it's a bit easier to reason about by breaking the steps down, imo